lib/home_page.dart

```diff
@@ -8,15 +24,288 @@ class HomePage extends StatefulWidget {
+          });
+        }
+      },
+    ).subscribe();
+    SupabaseManager.supabaseClient.channel('public:ride_events').on(
+      RealtimeListenTypes.postgresChanges,
+      ChannelFilter(event: 'INSERT', schema: 'public', table: 'ride_events'), //, filter: 'sender_id=neq.$profileId'
+      // ignore: always_specify_types
+      (payload, [ref]) {
+        // ignore: avoid_dynamic_calls
+        if (payload['new']['sender_id'] != profileId) {
+          // ignore: avoid_dynamic_calls
+          _loadNewRideEvent(payload['new']['id']);
+        }
+      },
+    ).on(
+      RealtimeListenTypes.postgresChanges,
+      ChannelFilter(event: 'UPDATE', schema: 'public', table: 'ride_events'), //, filter: 'sender_id=neq.$profileId'
+      // ignore: always_specify_types
+      (payload, [ref]) {
+        // ignore: avoid_dynamic_calls
+        if (payload['new']['read'] == true) {
+          setState(() {
+            // ignore: avoid_dynamic_calls
+            _items = _items.where((Model element) => element is Message || element.id != payload['new']['id']).toList();
+          });
+        }
+      },
+    ).subscribe();
+
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    SupabaseManager.supabaseClient.removeAllChannels();
+    super.dispose();
+  }
+
+  Future<void> load() async {
+    final List<Model> items = <Model>[];
+    final int profileId = SupabaseManager.getCurrentProfile()!.id!;
+    final List<Map<String, dynamic>> messagesData = parseHelper.parseListOfMaps(
+      await SupabaseManager.supabaseClient.from('messages').select('''
+      *,
+      sender: sender_id(*)
+      )
+    ''').eq('read', false).neq('sender_id', profileId).order('created_at'),
+    );
+    items.addAll(Message.fromJsonList(messagesData));
+    final List<Map<String, dynamic>> rideEventsData = parseHelper.parseListOfMaps(
+      await SupabaseManager.supabaseClient.from('ride_events').select('''
+      *,
+      ride: ride_id(*,
+        rider: rider_id(*),
+        drive: drive_id(*,
+          driver: driver_id(*)
+          )
+      )
+      ''').eq('read', false).order('created_at'),
+    );
+    items.addAll(RideEvent.fromJsonList(rideEventsData).where((RideEvent rideEvent) => rideEvent.isForCurrentUser()));
+    items.sort((Model a, Model b) => b.createdAt!.compareTo(a.createdAt!));
+
+    setState(() {
+      _items = items;
+      _fullyLoaded = true;
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
-        title: const Text('Home'),
+        title: Text(S.of(context).pageHomeTitle),
       ),
-      body: const Center(
-        child: Text('Home'),
+      body: Padding(
+        padding: const EdgeInsets.symmetric(
+          horizontal: 20,
+          vertical: 10,
+        ),
+        child: Column(
+          children: <Widget>[
+            const SizedBox(height: 30),
+            if (_fullyLoaded)
+              Expanded(
+                child: Container(
```

Comment:
I don't like this container. I think it could be worth a try putting the buttons at the top and make them scroll along with the events. If we want to keep the buttons in sight at all times, we can try a little CustomScrollView/PersistentHeader magic.

Reply:
He means the outline I think. And I don't know how it could look good in any way without the outline. Imo it's definitely needed.

Reply:
There has to be a way to make this look better, and with the current design we will be running into the problems with upcoming trips I described below... But that can be addressed in another PR I suppose